src/emd/rt_bse_types.F

```diff
@@ -0,0 +1,791 @@
+!> \param sim_time Current time of the simulation
+!> \param sim_dt Timestep of the simulation
+!> \param etrs_threshold Self-consistency threshold for enforced time reversal symmetry propagation
+!> \param exp_accuracy Threshold for matrix exponential calculation
+!> \param dft_control DFT control parameters
+!> \param ham_effective Real and imaginary part of the effective Hamiltonian used to propagate
+!>                      the density matrix
+!> \param ham_reference Reference Hamiltonian, which does not change in the propagation = DFT+G0W0 - initial Hartree - initial COHSEX
+!> \param ham_workspace Workspace matrices for use with the Hamiltonian propagation - storage of
+!>                      exponential propagators etc.
+!> \param rho Density matrix at the current time step
+!> \param rho_new Density matrix - workspace in ETRS
+!> \param rho_last Density matrix - workspace in ETRS
+!> \param rho_new_last Density matrix - workspace in ETRS
+!> \param rho_M Density matrix - workspace in ETRS
+!> \param S_inv_fm Inverse overlap matrix, full matrix
+!> \param S_fm Overlap matrix, full matrix
+!> \param S_inv Inverse overlap matrix, sparse matrix
+!> \param rho_dbcsr Density matrix, sparse matrix
+!> \param rho_workspace Matrices for storage of density matrix at different timesteps for
+!>                      interpolation and self-consistency checks etc.
+!> \param complex_workspace Workspace for complex density (exact diagonalisation)
+!> \param complex_s Complex overlap matrix (exact diagonalisation)
+!> \param real_eigvals Eigenvalues of hermitian matrix (exact diagonalisation)
+!> \param exp_eigvals Exponentiated eigenvalues (exact diagonalisation)
+!> \param v_dbcsr Sparse matrix with bare Coulomb in RI basis
+!> \param w_dbcsr Sparse matrix with correlation part of dressed Coulomb in RI basis (without bare Coulomb)
+!> \param screened_dbt Tensor for screened Coulomb interaction
+!> \param sigma_dbt Tensor for self-energy
+!> \param greens_dbt Tensor for greens function/density matrix
+!> \param t_3c_w Tensor containing 3c integrals
+!> \param t_3c_work_RI__AO_AO Tensor sigma contraction
+!> \param t_3c_work_RI_AO__AO Tensor sigma contraction
+!> \param t_3c_work2_RI_AO__AO Tensor sigma contraction
+!> \param sigma_SEX Screened exchange self-energy
+!> \param sigma_COH Coulomb hole self-energy
+!> \param hartree_curr Current Hartree matrix
+!> \param etrs_max_iter Maximum number of ETRS iterations
+!> \param ham_reference_type Which Hamiltonian to use as single particle basis
+!> \param mat_exp_method Which method to use for matrix exponentiation
+!> \param unit_nr Number of output unit
+!> \param int_3c_array Array containing the local 3c integrals
+!> \author Stepan Marek (01.24)
+! **************************************************************************************************
+   TYPE rtbse_env_type
+      INTEGER                                                   :: n_spin, &
```

Comment:
Provide defaults for all of its components

Reply:
I provided defaults for the components